File: python/roman1.py
# accept integer from user
# check if number is between 1 and 10 inclusive
# if number is between 1 and 10 inclusive, convert to roman numeral
# else print error message

num = int(input("Enter a number between 1 and 10: "))

roman = {
    1: "I",
    2: "II",
    3: "III",
    4: "IV",
    5: "V",
    6: "VI",
    7: "VII",
    8: "VIII",
    9: "IX",
    10: 'X'
}

if num in roman:
    print(roman[num])
else:
    print("Error: Number is not between 1 and 10")

File: java/roman1.java
public class roman {
    int num = 0;
    String romanNum = "";
    public static void main(String[] args) {
        roman r = new roman();
        r.getInt();
        r.checkNum();
        r.convertNum();
        System.out.println(r.romanNum);
    }

    public void getInt() {
        try {
            System.out.print("Enter a non-zero positive integer: ");
            num = Integer.parseInt(System.console().readLine());
        } catch (NumberFormatException e) {
            System.out.println("Invalid input. Please enter a non-zero positive integer.");
            getInt();
        }
    }

    public void checkNum() {
        if (num < 1 || num > 10) {
            System.out.println("Invalid input. Please enter a non-zero positive integer.");
        }
    }

    public void convertNum() {
        if (num == 1) {
            romanNum = "I";
        } else if (num == 2) {
            romanNum = "II";
        } else if (num == 3) {
            romanNum = "III";
        } else if (num == 4) {
            romanNum = "IV";
        } else if (num == 5) {
            romanNum = "V";
        } else if (num == 6) {
            romanNum = "VI";
        } else if (num == 7) {
            romanNum = "VII";
        } else if (num == 8) {
            romanNum = "VIII";
        } else if (num == 9) {
            romanNum = "IX";
        } else if (num == 10) {
            romanNum = "X";
        }
    }

    public String getRomanNum() {
        return romanNum;
    }


}